### PR TITLE
stats: extended maximum number of items in the statistics map

### PR DESCRIPTION
**Pull request by kapi-no**
Drivers, Flash simulator

Extended the maximum possible number of sections that track statistics
for each flash page.

This was previously limited by:
- UTIL_REPEAT macro
- stats.h API

### CODE REVIEW

**Review by nvlsianpu — APPROVED** [review]

**Comment by nvlsianpu** [comment]
cc @ccollins476ad

File: include/stats.h [hunk 1]
```diff
@@ -214,8 +214,8 @@ struct stats_hdr {
  * @param group__               The group containing the entry to clear.
  * @param var__                 The statistic entry to clear.
  */
-void stats_init(struct stats_hdr *shdr, uint8_t size, uint8_t cnt,
-		const struct stats_name_map *map, uint8_t map_cnt);
+void stats_init(struct stats_hdr *shdr, uint8_t size, uint16_t cnt,
```
Comment (jukkar) [inline include/stats.h]: Shouldn't we have `u16_t` here?

File: include/stats.h [hunk 2]
```diff
@@ -255,8 +255,8 @@ int stats_register(const char *name, struct stats_hdr *shdr);
  *
  * @see STATS_INIT_AND_REG
  */
-int stats_init_and_reg(struct stats_hdr *hdr, uint8_t size, uint8_t cnt,
-		       const struct stats_name_map *map, uint8_t map_cnt,
+int stats_init_and_reg(struct stats_hdr *hdr, uint8_t size, uint16_t cnt,
```
Comment (jukkar) [inline include/stats.h]: and here

File: include/stats.h (hunk 1 above)
Comment (kapi-no) [reply include/stats.h]: We probably should, but I wanted to keep existing conventions. I think that type alignment for stats.h should be done in a separate PR/commit

**Comment by kapi-no** [comment]
@nashif or @carlescufi Could you merge?

File: include/stats.h (hunk 1 above)
Comment (nvlsianpu) [reply include/stats.h]: @jukkar migration form stdint.h is out of scope his PR. Can we kept this as is?

Comment (jukkar) [reply include/stats.h]: Well, it just looks weird that the header file has uint8_t and uint16_t vars, but the .c file uses u8_t and u16_t. In this respect I would say this needs to change.

File: include/misc/util.h [hunk 3]
```diff
@@ -373,6 +373,222 @@ static inline s64_t arithmetic_shift_right(s64_t value, u8_t shift)
 #define UTIL_DEC_38 37
 #define UTIL_DEC_39 38
 #define UTIL_DEC_40 39
+#define UTIL_DEC_41 40
```
Comment (galak) [inline include/misc/util.h]: Can this be reworked to not require a #define at all?

Comment (kapi-no) [reply include/misc/util.h]: Could you suggest something specific? How should I go about this?

File: include/stats.h (hunk 1 above)
Comment (kapi-no) [reply include/stats.h]: @jukkar, Stats API now uses Zephyr types

File: include/stats.h (hunk 2 above)
Comment (kapi-no) [reply include/stats.h]: fixed

File: include/misc/util.h (hunk 3 above)
Comment (kapi-no) [reply include/misc/util.h]: Could you refactor it in independent PR? It seems you have a clear idea of how to approach this.

**Comment by kapi-no** [comment]
@galak or @jukkar Could you merge this?

**Review by mbolivar — APPROVED** [review]
Looks sane to me. I think the suggested util.h changes aren't in scope here; it's just extending existing UTIL_REPEAT() functionality as-is, rather than changing how it is implemented in the first place.

**Pull request merged**